Packs/ZeroNetworksSegment/Integrations/ZeroNetworksSegmentEventCollector/ZeroNetworksSegmentEventCollector.py

```diff
@@ -0,0 +1,433 @@
+    When 'ok' is returned it indicates the integration works like it is supposed to and connection to the service is
+    successful.
+    Raises exceptions if something goes wrong.
+    Args:
+        client (Client): ZeroNetworks client to use.
+    Returns:
+        str: 'ok' if test passed, anything else will raise an exception and will fail the test.
+    """
+
+    try:
+        start_timestamp = initialize_start_timestamp({}, "")
+        client.search_events(limit=1, cursor=start_timestamp, log_type=AUDIT_TYPE)
+    except Exception as e:
+        if 'Unauthorized' in str(e):
+            return 'Authorization Error: make sure the API Key is correctly set'
+        else:
+            raise e
+    return 'ok'
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:
+    """
+    Main function for parsing parameters and executing command functions.
+    """
+    params = demisto.params()
+    args = demisto.args()
+    command = demisto.command()
+
+    api_key = params.get('credentials', {}).get('password')
+    server_url = urljoin(params.get('url'))
+    verify_certificate = not argToBoolean(params.get('insecure', False))
+    proxy = params.get('proxy', False)
+    should_push_events = argToBoolean(args.get('should_push_events', False))
+    event_types_to_fetch = argToList(params.get('event_types_to_fetch', []))
+    log_types = handle_log_types(event_types_to_fetch)
+    isFetch = params.get('isFetchEvents')
+    if isFetch and not log_types:
+        raise DemistoException("At least one event type must be specified for fetching.")
+
+    demisto.debug(f'Event types that will be fetched in this instance: {log_types}')
+
+    filters = params.get('network_activity_filters', '')
+    if 'network_activities' in log_types and not filters:
+        raise DemistoException(
+            "Using network_activity_filters is required when fetching network events, to limit the number of events.")
+
+    demisto.debug(f'Command being called is {command}')
+    try:
+        headers: dict = {'accept': 'application/json', 'Authorization': api_key}
+
+        client = Client(
+            server_url=server_url,
+            proxy=proxy,
+            verify=verify_certificate,
+            headers=headers)
+
+        if command == 'test-module':
+            # This is the call made when pressing the integration Test button.
+            return_results(test_module(client))
+
+        elif command == 'zero-networks-segment-get-events':
+            last_run = demisto.getLastRun()
+            events, results = get_events(client, args, last_run, params, log_types)  # type: ignore
+            return_results(results)
+            if should_push_events:
+                send_events_to_xsiam(events, vendor=VENDOR, product=PRODUCT)
+
+        elif command == 'fetch-events':
+            last_run = demisto.getLastRun() or {}
+            next_run, events = fetch_all_events(client, params, last_run, log_types)
+            demisto.setLastRun(next_run)
```

Comment:
Set last run needs to be after send_events_to_xsiam since if the function will fail and not send the events you already moved the progressed the last_run